src/components/BulkUploadListItem.tsx

```diff
@@ -22,112 +22,95 @@ export const BulkUploadListItem = ({ upload }: BulkUploadListItemProps) => {
-    }
-
-    if (isInProgress) {
-      return <ArrowPathIcon className="icon rotating" />;
-    }
-
-    if (isCompleted) {
-      return <CheckCircleIcon className="icon" />;
-    }
-
-    if (isCanceled || isFailed) {
-      return <XCircleIcon className="icon" />;
-    }
-
-    return null;
-  };
-
-  const makeHumanizedTime = () => {
-    const relativizedTime = relativizeDateTime(createdAt);
-    const localizedTime = localizeDateTime(createdAt);
-
-    return (
-      <time dateTime={createdAt} title={localizedTime}>{relativizedTime}</time>
-    );
-  };
-
-  const makeMetadata = () => {
-    const humanizedTimestamp = makeHumanizedTime();
-
-    if (isPending) {
-      return (
-        <>
-          {humanizedTimestamp}
-          {METADATA_DELIMITER}
-          Waiting to process…
-        </>
-      );
-    }
+  const humanizedTimestamp = (
+    <time dateTime={createdAt} title={localizeDateTime(createdAt)}>
+      {relativizeDateTime(createdAt)}
+    </time>
+  );
 
-    if (isInProgress) {
-      return (
-        <>
-          {humanizedTimestamp}
-          {METADATA_DELIMITER}
-          Processing…
-        </>
-      );
-    }
+  let icon = null;
+  let metadata = [];
 
-    if (isCompleted) {
-      // TODO: Report how many rows were processed. (See mockups in #413.)
-      return humanizedTimestamp;
-    }
+  // Don't let the bullet wrap to the beginning of a new line.
+  const METADATA_DELIMITER = <>&nbsp;• </>;
 
-    if (isCanceled) {
-      return (
+  if (isPending) {
+    icon = <ClockIcon className="icon" />;
+    metadata.push(humanizedTimestamp, 'Waiting to process…');
+  }
+
+  if (isInProgress) {
+    icon = <ArrowPathIcon className="icon rotating" />;
+    metadata.push(humanizedTimestamp, 'Processing…');
+  }
+
+  if (isCompleted) {
+    icon = <CheckCircleIcon className="icon" />;
+    metadata.push(humanizedTimestamp);
+  }
+
+  if (isCanceled) {
+    icon = <XCircleIcon className="icon" />;
+    metadata.push(
+      (
         <>
           Canceled
           {' '}
           {humanizedTimestamp}
         </>
-      );
-    }
+      )
+    );
+  }
 
-    if (isFailed) {
-      return (
+  if (isFailed) {
+    icon = <XCircleIcon className="icon" />;
+    metadata.push(
+      (
         <>
           Failed
           {' '}
           {humanizedTimestamp}
-          {METADATA_DELIMITER}
-          <EmailLink
-            to="info@philanthropydatacommons.org"
-            subject={`Bulk upload error (ID: ${id})`}
-          >
-            Email the dev team for help
-          </EmailLink>
         </>
-      );
-    }
+      ),
+      (
+        <EmailLink
+          to="info@philanthropydatacommons.org"
+          subject={`Bulk upload error (ID: ${id})`}
+        >
+          Email the dev team for help
+        </EmailLink>
+      )
+    );
+  }
 
-    return null;
-  };
+  metadata = metadata.filter((datum) => datum).map((datum, i) => (
+    // eslint-disable-next-line react/no-array-index-key
```

Comment:
Another instance of array index being used as a key; does `datum` have any type of identifier?

Reply:
It does not, unfortunately. I could probably refactor it to have that by making `metadata` an object instead of an array, so we can use the object keys as part of the key construction… I'll try that!